File: app/src/main/java/net/daverix/ajvm/jvm/FStoreOperation.java
package net.daverix.ajvm.jvm;


import net.daverix.ajvm.Frame;
import net.daverix.ajvm.ByteCodeReader;

import java.io.IOException;

public class FStoreOperation implements ByteCodeOperation {
    @Override
    public void execute(ByteCodeReader reader,
                        int indexOfBytecode,
                        Frame currentFrame) throws IOException {
        int index = reader.readUnsignedByte();
        Object localVariable = currentFrame.pop();

        if(!(localVariable instanceof Float))
            throw new IllegalStateException("variable " + localVariable + " is not an Float");

        currentFrame.setLocalVariable(index, localVariable);
    }
}

File: app/src/main/java/net/daverix/ajvm/operation/FStoreOperation.kt
package net.daverix.ajvm.operation


import net.daverix.ajvm.Frame
import net.daverix.ajvm.ByteCodeReader

import java.io.IOException

class FStoreOperation : ByteCodeOperation {
    @Throws(IOException::class)
    override fun execute(reader: ByteCodeReader,
                         indexOfBytecode: Int,
                         currentFrame: Frame) {
        val index = reader.readUnsignedByte()
        val localVariable = currentFrame.pop()

        if (localVariable !is Float)
            throw IllegalStateException("variable $localVariable is not an Float")

        currentFrame.setLocalVariable(index, localVariable)
    }
}
